src/aggregation/agg_req_with_accessor.rs

```diff
@@ -50,51 +53,51 @@ impl AggregationWithAccessor {
         limits: AggregationLimits,
     ) -> crate::Result<AggregationWithAccessor> {
         let mut str_dict_column = None;
+        let mut accessor2 = None;
         use AggregationVariants::*;
         let (accessor, field_type) = match &agg.agg {
             Range(RangeAggregation {
                 field: field_name, ..
-            }) => get_ff_reader_and_validate(
-                reader,
-                field_name,
-                Some(get_numeric_or_date_column_types()),
-            )?,
+            }) => get_ff_reader(reader, field_name, Some(get_numeric_or_date_column_types()))?,
             Histogram(HistogramAggregation {
                 field: field_name, ..
-            }) => get_ff_reader_and_validate(
-                reader,
-                field_name,
-                Some(get_numeric_or_date_column_types()),
-            )?,
+            }) => get_ff_reader(reader, field_name, Some(get_numeric_or_date_column_types()))?,
             DateHistogram(DateHistogramAggregationReq {
                 field: field_name, ..
-            }) => get_ff_reader_and_validate(
-                reader,
-                field_name,
-                Some(get_numeric_or_date_column_types()),
-            )?,
+            }) => get_ff_reader(reader, field_name, Some(get_numeric_or_date_column_types()))?,
             Terms(TermsAggregation {
                 field: field_name, ..
             }) => {
                 str_dict_column = reader.fast_fields().str(field_name)?;
-                get_ff_reader_and_validate(reader, field_name, None)?
+                let allowed_column_types = [
+                    ColumnType::I64,
+                    ColumnType::U64,
+                    ColumnType::F64,
+                    ColumnType::Bytes,
+                    ColumnType::Str,
+                    // ColumnType::Bool Unsupported
```

Comment:
Can we add support for those? Is there a ticket for this?

Reply:
There's a ticket
https://github.com/quickwit-oss/tantivy/issues/1689

We can support them all, IP requires some API extensions on columnar